packages/mdl-drawer/temporary/foundation.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings, numbers} from './constants';
+
+export default class MDLTemporaryDrawerFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get numbers() {
+    return numbers;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hasClass: (/* className: string */) => {},
+      hasNecessaryDOM: () => /* boolean */ {},
+      registerComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerTransitionEndHandler: (/* handler: EventListener */) => {},
+      deregisterTransitionEndHandler: (/* handler: EventListener */) => {},
+      updateDrawerPosition: (/* value: number | null */) => {},
+      updateComponentBackground: (/* value: string */) => {},
+      getFocusableElements: () => /* array<Element> */ {},
+      saveElementTabState: (/* el: Element */) => {},
+      restoreElementTabState: (/* el: Element */) => {},
+      makeElementUntabbable: (/* el: Element */) => {},
+      isRightSide: () => /* boolean */ {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDLTemporaryDrawerFoundation.defaultAdapter, adapter));
+
+    this.transitionEndHandler_ = () => {
+      this.adapter_.removeClass(MDLTemporaryDrawerFoundation.cssClasses.ANIMATING);
+      this.adapter_.deregisterTransitionEndHandler(this.transitionEndHandler_);
+    };
+
+    this.inert_ = false;
+
+    this.componentClickHandler_ = () => this.close();
+    this.drawerClickHandler_ = evt => evt.stopPropagation();
+    this.componentTouchStartHandler_ = evt => this.handleTouchStart_(evt);
+    this.componentTouchMoveHandler_ = evt => this.handleTouchMove_(evt);
+    this.componentTouchEndHandler_ = evt => this.handleTouchEnd_(evt);
+  }
+
+  init() {
+    const {ROOT, OPEN} = MDLTemporaryDrawerFoundation.cssClasses;
+
+    if (!this.adapter_.hasClass(ROOT)) {
```

Comment:
Hmm. The downside of this is that it could lead to code bloat if used extensively, especially in production where (hopefully) developers have already addressed these issue.

Do you think it's worth investigating this pattern a bit more, and then looking at solutions such as creating different "development" vs. "production" builds, or having an external tool that provides this validation separately from the core lib?